### DISCUSSION TITLE
Return Row Selected Row Data

### DISCUSSION THREAD

**Discussion by BoSmith** [yajra/laravel-datatables#2962] [Q&A]
This seems like it would be a simple task but I cannot find documentation on how to get the data from a row selected in a datatable. I would like to have either a button that "submits" the selected row data back to the laravel controller for processing or even better yet just being able to double click a row would be better to get the row's data returned to the server. 
Thanks for the help.

**Comment by yajra** [comment]
I think what you need is the [select extension](https://datatables.net/extensions/select/examples/).

**Reply by bigdogdman** [reply]
@yajra, how do I integrate the select extension into laravel? On that note, how do I integrate _any_ datatable extensions into laravel? It seems to me because of how laravel datatables works it needs to be 100% configurable before being drawn the first time, and I'm having difficulty locating all the documentation for all of that config for the different functionalities?

**Reply by yajra** [reply]
Integrating scripts depends on your build. 

1. You can integrate via npm and then use laravel-mix or vite.
2. Some templates already have dataTables bundled with them.
3. You do it traditionally by downloading the files in datatables.net and adding them to your HTML.

You can follow the quick guide I made https://yajrabox.com/docs/laravel-datatables/10.0/quick-starter if you haven't tried it yet.

Thanks!

**Reply by bigdogdman** [reply]
I have tried it, thanks, but I clearly did it wrong, as is evidenced in #3147. But I'm going to hold off on this stuff until I get straightened out on _that_ one.
...Except for maybe the install...
So are there any built-in extensions in yajra datatables? In your quick-starter you use Buttons and exporting and all that jazz, so...does that mean that stuff's already included in yours?

**Reply by yajra** [reply]
Yes, there are built-in custom buttons aside from the one provided by datatables.net. See https://github.com/yajra/laravel-datatables-vite

If you have the editor license, maybe this can help https://github.com/yajra/laravel-datatables-tailwind-experiment with the js setup. 

Or follow this file https://github.com/yajra/laravel-datatables-tailwind-experiment/blob/main/resources/js/dataTables.tailwindcss.js

**Reply by bigdogdman** [reply]
Wow, lots to research...thanks for the links! I'll definitely give them a look-see. :smiley:
